src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -51,15 +51,14 @@ public Product findProductByIdOrThrowException(UUID productId) {
     }
 
     @Override
-    public void deleteProduct(UUID productId){
+    public ProductResponse deleteProduct(UUID productId){
         ifProductIsNotDeletedThrowException(productId);
         Product product = findProductByIdOrThrowException(productId);
```

Comment:
Con lo que te marqué abajo esto podría quedar

```suggestion
        Product product = findProductByIdOrThrowException(productId);
        ifProductIsNotDeletedThrowException(product);
```

Y no somos redundantes